src/classes/DRS_ContactMerge_CTRL

```diff
@@ -0,0 +1,1008 @@
+                    }
+                }
+                // Collect the standard and custom fields separately for sorting
+                if(fResult.isCustom()) {
+                    customFieldMap.put(fieldName, fResult);
+                } else {
+                    standardFieldMap.put(fieldName, fResult);
+                }
+            }
+        }
+        // Adding some non-updateable system fields which we need to add to the record diff table.
+        query +=  'createdby.name, createddate, LastModifiedBy.name, LastModifiedDate';
+        // Finally completing the query by appending the table name and the filter clause
+        query += ' from Contact where id IN :mergeCandidates';
+
+        //UTIL_Debug.debug('The contact query is: ' + query);
+
+        List<Contact> contacts;
+        try {
+            contacts = Database.query(query); // Query the records
+            // Ensure we got back the same number of records as expected. In case any record got deleted/moved since last search.
+            if (contacts == null || contacts.size() != mergeCandidates.size()) {
+                ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, Label.npe01.Contact_Merge_Error_Query_Failed + ' ' + Label.npe01.Contact_Merge_Error_please_retry));
+                return;
+            }
+        } catch (Exception e) {
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, Label.npe01.Contact_Merge_Error_Contact_not_found + ' Error: ' + e));
+            return;
+        }
+
+        if (!canUserDeleteSelectedContacts(mergeCandidates)) {
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, System.Label.conMergeErrorNoDeleteRecAccess));
+            return;
+        }
+
+        // Update the list of searched records with only the ones selected
+        contactResults.clear();
+        for (Contact c : contacts) {
+            this.selectedRecords.put(c.id, c);
+        }
+
+        // Sort the standard fields list by name before creating the list of FieldRow for merge UI
+        List<String> standardFieldList = new List<String>();
+        standardFieldList.addAll(standardFieldMap.keySet());
+        standardFieldList.sort();
+
+        // Sort the custom fields list by name before creating the list of FieldRow for merge UI
+        List<String> customFieldList = new List<String>();
+        customFieldList.addAll(customFieldMap.keySet());
+        customFieldList.sort();
+
+        // Add the first row of type header with contact names as titles
+        FieldRow temp = new FieldRow('', '', false, 'header');
+        for (Sobject c: contacts) {
+            Contact con = (Contact) c;
+            temp.values.add(new Cell(c.id, con.name));
+        }
+        fieldRows.add(temp);
+
+        // Add second row to select the master record. Select the one which is last modified
+        temp = new FieldRow(MASTER_KEY, 'Master Record', true, null);
+        {
```

Comment:
Can remove this set of curly braces (line #771 and 781). I don't *think* they are doing anything

Reply:
Updated